### PR TITLE
add parent ID migrator for notion nodes

### PR DESCRIPTION
**Pull request by fontanierh**

### CODE REVIEW

File: front/migrations/20241211_parents_migrator.ts [hunk 1]
```diff
@@ -143,7 +144,24 @@ const migrators: Record<ConnectorProvider, ProviderMigrator | null> = {
...
+        // legacy parents
+        ...uniqueIds,
+        // new parents
+        ...uniqueIds.map((id) => `notion-${id}`),
+      ];
+    },
+    cleaner: (nodeId, parents) =>
+      // Only keep the new parents
+      _.uniq([nodeId, ...parents].map((x) => _.last(x.split("notion-"))!)).map(
+        (id) => `notion-${id}`
```
Comment (aubin-tchoi) [inline front/migrations/20241211_parents_migrator.ts:162]: love that 👍

**Review by aubin-tchoi — APPROVED** [review]
LGTM

File: front/migrations/20241211_parents_migrator.ts [hunk 2]
```diff
@@ -143,7 +144,24 @@ const migrators: Record<ConnectorProvider, ProviderMigrator | null> = {
...
+  notion: {
+    transformer: (nodeId, parents) => {
+      const uniqueIds = _.uniq(
+        [nodeId, ...parents].map((x) => _.last(x.split("notion-"))!)
+      );
+      return [
+        // legacy parents
+        ...uniqueIds,
+        // new parents
+        ...uniqueIds.map((id) => `notion-${id}`),
```
Comment (spolu) [inline front/migrations/20241211_parents_migrator.ts:156]: nodeId is in uniqueId here?

Comment (fontanierh) [reply front/migrations/20241211_parents_migrator.ts:156]: not quite sure what you mean

File: front/migrations/20241211_parents_migrator.ts [hunk 3]
```diff
@@ -143,7 +144,24 @@ const migrators: Record<ConnectorProvider, ProviderMigrator | null> = {
   google_drive: null,
   microsoft: null,
   github: null,
-  notion: null,
+  notion: {
+    transformer: (nodeId, parents) => {
```
Comment (spolu) [inline front/migrations/20241211_parents_migrator.ts:148]: this one is not idempotent if it fails at some point you're in trouble right?

Comment (fontanierh) [reply front/migrations/20241211_parents_migrator.ts:148]: it's quite idempotent

Comment (spolu) [reply front/migrations/20241211_parents_migrator.ts:148]: Agreed

**Review by spolu — APPROVED** [review]
LGTM modulo making sure that nodeId = parents[0] post transform

**Comment by fontanierh** [comment]
> LGTM modulo making sure that nodeId = parents[0] post transform

Yep 👍

**Pull request merged**